dataflow/src/main/java/org/checkerframework/dataflow/reachingdef/ReachingDefinitionStore.java

```diff
@@ -0,0 +1,119 @@
+
+    /** Create a new ReachDefinitionStore. */
+    public ReachingDefinitionStore() {
+        reachingDefSet = new LinkedHashSet<>();
+    }
+
+    /**
+     * Create a new ReachDefinitionStore.
+     *
+     * @param reachingDefSet a set of reaching definition nodes, this parameter is captured and
+     *     that the caller should not retain an alias
+     */
+    public ReachingDefinitionStore(LinkedHashSet<ReachingDefinitionNode> reachingDefSet) {
+        this.reachingDefSet = reachingDefSet;
+    }
+
+    /**
+     * Remove the information of a reaching definition from the reaching definition set.
+     *
+     * @param defTarget target of a reaching definition
+     */
+    public void killDef(Node defTarget) {
+        Iterator<ReachingDefinitionNode> it = reachingDefSet.iterator();
+        while (it.hasNext()) {
+            // We use `.equals` instead of `==` here to compare value equality
+            // rather than reference equality, because if two left-hand side node
+            // have same values, we need to kill the old one and replace with the
+            // new one.
+            ReachingDefinitionNode generatedDefNode = it.next();
+            if (generatedDefNode.def.getTarget().equals(defTarget)) {
+                it.remove();
+            }
+        }
+    }
+
+    /**
+     * Add a reaching definition to the reaching definition set.
+     *
+     * @param def a reaching definition
+     */
+    public void putDef(ReachingDefinitionNode def) {
+        reachingDefSet.add(def);
+    }
+
+    @Override
+    public boolean equals(@Nullable Object obj) {
+        if (!(obj instanceof ReachingDefinitionStore)) {
+            return false;
+        }
+        ReachingDefinitionStore other = (ReachingDefinitionStore) obj;
+        return other.reachingDefSet.equals(this.reachingDefSet);
+    }
+
+    @Override
+    public int hashCode() {
+        return this.reachingDefSet.hashCode();
+    }
+
+    @Override
+    public ReachingDefinitionStore copy() {
+        return new ReachingDefinitionStore(new LinkedHashSet<>(reachingDefSet));
+    }
+
+    @Override
+    public ReachingDefinitionStore leastUpperBound(ReachingDefinitionStore other) {
+        LinkedHashSet<ReachingDefinitionNode> reachingDefSetLub =
+                new LinkedHashSet<>(this.reachingDefSet.size() + other.reachingDefSet.size());
+        reachingDefSetLub.addAll(this.reachingDefSet);
+        reachingDefSetLub.addAll(other.reachingDefSet);
+        return new ReachingDefinitionStore(reachingDefSetLub);
+    }
+
+    @Override
+    public ReachingDefinitionStore widenedUpperBound(ReachingDefinitionStore previous) {
+        throw new BugInCF("wub of reaching definition get called!");
+    }
+
+    @Override
+    public boolean canAlias(JavaExpression a, JavaExpression b) {
+        return true;
+    }
+
+    @Override
+    public String visualize(CFGVisualizer<?, ReachingDefinitionStore, ?> viz) {
+        String key = "reaching definitions";
```

Comment:
Is it possible that the pdf for the manual wasn't updated?
There I see `reach definitions = a = 1, b =2`.
The sequence of `=` looks a bit confusing.
Would `reaching definitions = {a = 1, b =2}` look better?
You should get that by changing to `new StringJoiner(", ", "{ ", " }")` below.
Maybe we should do something similar for the other analyses, in separate PRs.

Reply:
Yes, let me update the pdf for the manual.

Sure, using {} is easier to look.